090-integrate-bridge-with-metrics-reporter.md

```diff
@@ -0,0 +1,113 @@
+# Integrate Bridge with Metrics Reporter
+
+In [SIP-064](https://github.com/strimzi/proposals/blob/main/064-prometheus-metrics-reporter.md) we introduced the [Strimzi Metrics Reporter](https://github.com/strimzi/metrics-reporter).
+This is a Kafka `MetricsReporter` plugin that directly exposes metrics in Prometheus format via an HTTP endpoint.
+
+## Current situation
+
+The Cluster Operator exposes Kafka metrics through the [Prometheus JMX Exporter](https://github.com/prometheus/jmx_exporter), which can be configured using the shared `metricsConfig` schema.
+This schema has a `type` property that only allows the `jmxPrometheusExporter` value, and a reference to a `ConfigMap` containing its configuration.
+
+```sh
+metricsConfig:
+  type: jmxPrometheusExporter
+  valueFrom:
+    configMapKeyRef:
+      name: <component>-metrics
+      key: metrics-config.yml
+```
+
+At runtime, this configuration enables a Java agent that exposes Kafka JMX metrics in Prometheus format through an HTTP endpoint on port 9404.
+Note that this agent depends on the [Kafka JMX Reporter](https://github.com/apache/kafka/blob/3.9.0/clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java) plugin, which is enabled by default.
+
+The only major component not using the `metricsConfig` schema is the [HTTP Bridge](https://github.com/strimzi/strimzi-kafka-bridge), as it directly instantiates the JMX Exporter's `JmxCollector` class with hard coded configuration.
+The `KafkaBridge` CRD does not support a type system for metrics configuration, but a simple boolean configuration to enable/disable metrics.
+
+```sh
+spec:
+  enableMetrics: true
+```
+
+## Motivation
+
+We want to support the Strimzi Metrics Reporter as an alternative way of configuring metrics across all components.
+In order to provide a consistent user experience, the HTTP Bridge should align with all the other components in terms of metrics configuration. 
+
+## Proposal
+
+As specified in SIP-064, a new metrics config type called `strimziMetricsReporter` will be added to `metricsConfig` schema along with its configuration.
+Only the Strimzi Metrics Reporter's `prometheus.metrics.reporter.allowlist` property will be configurable by users when using the Cluster Operator.
+This configuration is a comma separated list of regex patterns that is used by the plugin to filter metrics to collect.
+All other plugin configuration properties will be locked down, including the port that will be set to 9404 to avoid conflicts with other ports.
+
+```sh
+spec:
+  kafka:
+    metricsConfig:
+      type: strimziMetricsReporter
+      values:
+        allowList:
+          - "kafka_log.*"
+          - "kafka_network.*"
+```
+
+### HTTP Bridge
+
+The HTTP Bridge project will be updated to support multiple types of metrics configurations.
+A new `bridge.metrics` property will be available, and will only accept `jmxPrometheusExporter` and `strimziMetricsReporter`.
```

Comment:
I would also expect that not defining it just defaults to the current behaviour by using JMX right? Should we state it here clearly?

Reply:
Current default is `KAFKA_BRIDGE_METRICS_ENABLED=false`, so when `bridge.metrics` will not be set, then the metrics endpoint will be disabled. This behavior is not changing.

Reply:
but we are deprecating the env var so it means that if you don't change the logic to default right now we have to do that in the future when the env var is not there anymore?

Reply:
Just to clarify what I mean ...
The implementation code will take care of KAFKA_BRIDGE_METRICS_ENABLED=false to understand that no metrics have to be exposed. But at some point the code should take care of "the bridge.metrics field is NOT set" to have the same (no metrics). So we should plan to switch to the second check when the env var is removed.

Reply:
This is already covered by the draft implementation. Let's discuss that when we get to it.